src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private final float loadFactor;
+    private int threshold;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        this.loadFactor = DEFAULT_LOAD_FACTOR;
+        this.threshold = (int) (loadFactor * DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = hash % table.length;
+        if (table[index] == null) {
+            table[index] = new Node<>(hash, key, value, null);
+            size++;
+        } else {
+            Node<K, V> currNode = table[index];
+            do {
+                if (currNode.hash == hash && Objects.equals(currNode.key, key)) {
+                    currNode.value = value;
+                    break;
+                }
+                if (currNode.next == null) {
+                    currNode.next = new Node<>(hash, key, value, null);
+                    size++;
+                    break;
+                }
+                currNode = currNode.next;
+            } while (true);
```

Comment:
Try to avoid `do while` loop. It's harder to read/understand.
```suggestion
            while (currNode != null) {
                if (currNode.hash == hash && Objects.equals(currNode.key, key)) {
                    currNode.value = value;
                    break;
                }
                if (currNode.next == null) {
                    currNode.next = new Node<>(hash, key, value, null);
                    size++;
                    break;
                }
                currNode = currNode.next;
            }
```